conda/plugins/environment_exporters/environment_yml.py

```diff
@@ -0,0 +1,100 @@
+# Copyright (C) 2012 Anaconda, Inc
+# SPDX-License-Identifier: BSD-3-Clause
+"""Built-in conda standard environment exporter plugins (YAML and JSON)."""
+
+from __future__ import annotations
+
+from typing import TYPE_CHECKING, Literal
+
+from ...common.serialize import json, yaml_safe_dump
+from ...exceptions import CondaValueError
+from ..hookspec import hookimpl
+from ..types import CondaEnvironmentExporter
+
+if TYPE_CHECKING:
+    from typing import Any, Final
+
+    from ...models.environment import Environment
+
+# Modern Python constants using Final annotations
+ENVIRONMENT_JSON_FORMAT: Final = "environment-json"
+ENVIRONMENT_YAML_FORMAT: Final = "environment-yaml"
+
+# Type alias for format validation
+EnvironmentFormatType = Literal["environment-json", "environment-yaml"]
+
+
+def to_dict(env: Environment) -> dict[str, Any]:
+    """
+    Convert Environment to standard dictionary format used by YAML and JSON exporters.
+
+    This represents the common dictionary structure that both YAML and JSON
+    environment formats use.
+
+    :param env: Environment model to convert
+    :return: Dictionary with standard environment fields
+    """
+    env_dict = {"name": env.name}
+
+    # Add channels if present
+    if env.config and env.config.channels:
+        env_dict["channels"] = list(env.config.channels)
+
+    # Convert requested_packages to dependencies list
+    if env.requested_packages:
+        # Use .spec to get space-separated format, then convert to single equals
+        env_dict["dependencies"] = [
+            requested_package.spec.replace(" ", "=")
+            for requested_package in env.requested_packages
+        ]
+    elif env.explicit_packages:
+        # Fall back to explicit packages if no requested packages
+        env_dict["dependencies"] = [
+            explicit_package.spec for explicit_package in env.explicit_packages
+        ]
+
+    # Add variables if present
+    if env.variables:
+        env_dict["variables"] = env.variables
+
+    # Add prefix if present (though this is less common in standard formats)
+    if env.prefix:
+        env_dict["prefix"] = env.prefix
+
+    return env_dict
+
+
+def export_yaml(env: Environment) -> str:
+    """Export Environment to YAML format."""
+    env_dict = to_dict(env)
+    yaml_content = yaml_safe_dump(env_dict)
```

Comment:
should we intercept YAML errors too the same way we do for JSON?

```suggestion
    try:
        yaml_content = yaml_safe_dump(env_dict))
    except ruamel.yaml.error.YAMLStreamError:
        raise CondaValueError(...)
```